cms/models.py

```diff
@@ -1071,6 +1113,149 @@ def get_context(self, request, *args, **kwargs):
         return context
 
 
+class SignatoryPage(Page):
+    """ CMS page representing a Signatory. """
+
+    promote_panels = []
+    parent_page_types = [SignatoryIndexPage]
+    subpage_types = []
+
+    name = models.CharField(
+        max_length=250, null=False, blank=False, help_text="Name of the signatory."
+    )
+    title_1 = models.CharField(
+        max_length=250,
+        null=True,
+        blank=True,
+        help_text="Specify signatory first title in organization.",
+    )
+    title_2 = models.CharField(
+        max_length=250,
+        null=True,
+        blank=True,
+        help_text="Specify signatory second title in organization.",
+    )
+    organization = models.CharField(
+        max_length=250,
+        null=True,
+        blank=True,
+        help_text="Specify the organization of signatory.",
+    )
+
+    image = models.ForeignKey(
+        Image,
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+        help_text="Signature image size must be at least 150x50 pixels.",
+    )
+
+    class Meta:
+        verbose_name = "Signatory"
+
+    content_panels = [
+        FieldPanel("name"),
+        FieldPanel("title_1"),
+        FieldPanel("title_2"),
+        FieldPanel("organization"),
+        ImageChooserPanel("image"),
+    ]
+
+    def save(self, *args, **kwargs):
+        # auto generate a unique slug so we don't hit a ValidationError
+        if not self.title:
+            self.title = self.__class__._meta.verbose_name.title() + "-" + self.name
+
+        self.slug = slugify("{}-{}-{}".format(self.title, self.name, self.id))
```

Comment:
`self.title` is already `{_meta.verbose_name.title}-{self.name}` so the slug will become `{verbose_name.title}-{self.name}-{self.name}-{self.id}`. Might want to remove `self.name` from `self.slug`.